pkg/config/msp.go

```diff
@@ -670,6 +670,23 @@ func (m *MSP) validateCACerts() error {
 	if err != nil {
 		return fmt.Errorf("invalid intermediate cert: %v", err)
 	}
+	rootCertPool := x509.NewCertPool()
+	for _, cert := range m.RootCerts {
+		rootCertPool.AddCert(cert)
+	}
+	//TODO: follow the workaround that msp code use to incorporate cert.Verify()
+	validIntermediateCert := false
+	for _, ic := range m.IntermediateCerts {
+		for _, rc := range m.RootCerts {
+			err := ic.CheckSignatureFrom(rc)
+			if err == nil {
+				validIntermediateCert = true
+			}
+		}
+	}
+	if !validIntermediateCert {
+		return errors.New("invalid intermediate cert")
+	}
```

Comment:
This logic only checks for a single valid intermediate cert. Not sure if you intended on validating all intermediate certs. If you want to you could return the error on 688 within the loop  if the err on 681 is not nil and if the loop passes all the way through you know all intermediate certs are valid.

Reply:
you are right. the original logic only checks if one intermediate cert is valid. fix applied.